src/ComunicaEngine.js

```diff
@@ -77,23 +77,29 @@ export default class ComunicaEngine {
     let sources = await source;
     if (!sources)
       return null;
+
     // Strip the fragment of a URI
     if (sources instanceof URL)
       sources = sources.href;
     else if (sources.termType === 'NamedNode')
       sources = sources.value;
+
     if (typeof sources === 'string')
       sources = [sources.replace(/#.*/, '')];
-    // Await multiple promises in an array
+    // Recurse arrays
     else if (Array.isArray(sources))
-      sources = await Promise.all(sources);
+      sources = [].concat(...(await Promise.all(sources.map(src => this.toComunicaSources(src)))));
```

Comment:
Just `[...()]` should do (but is there a reason we do the `...` thing in the first place)?

Reply:
Every Promise returns a call to toComunicaSources, which returns an array of results. So if we don't concat them we would get an array of arrays instead of an array of results. Which is also what we would get if we do `[...()]` unless I'm mistaken.